Book1/15/kernel regression attention

AnandShegde opened the pull request:
## Description
In the file book1/15/kernal_regression_attention.ipynb ModuleNotFoundError: module torchvision not found was popping up. Added a try except statement for torchvision.

<!-- Please refer to https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md before opening this PR -->

### Figure Number

<!-- Mention figure number with corresponding book (book1 or book2). For example: book1 - 4.18 -->

### Figures

<!-- Create a public gist with rendered figures and add a link below. 
Refer to this template for example: https://gist.github.com/patel-zeel/28a101009f5e938082b6d93101bf1fd7 -->

### Issue 
#759 
<!-- Link the issue you are solving -->

<!-- If the issue is from this repo, include directly with issue number -->
<!-- For example: 

#12

-->

<!-- If the issue is from another repo, you can include it using the regular linking mechanism -->
<!-- For example:

[Issue title](Issue link) 

-->

## Steps

<!-- If the task is complicated, you can specify each step using checkboxes so that everyone can understand what you have done and what you are going to complete. -->


- [x] Step1 added try except statement.

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

## Potential problems/Important remarks

<!-- If you have any important remarks for the reviewers to take a closer look, you can write them in detail here.  -->

karm-patel commented:
Dr. @murphyk , Currently d2l.py (originally) torch.py is coming from `https://raw.githubusercontent.com/d2l-ai/d2l-en/master/d2l/torch.py`, Soon we will transfer this file to `probml-utils`.

After transferring it to `probml-utils`, still we need to install `torchvision` in except block.  The reason is `d2l.py` is using torchvison and torch both  (attached snapshot), so ModuleNoFound error can be occurred due to missing `torch` or `tochvision`.
![image](https://user-images.githubusercontent.com/59387624/168737928-a4f19115-ea18-4bbe-b354-e156b8267572.png)

In that case code should be like this,
```py
try:
    from probml-utils import d2l
except ModuleNotFoundError:
    %pip install torch torchvision git+https://github.com/probml/probml-utils.git
    from probml-utils import d2l
```

karm-patel commented:
Now d2l.py is transferred to `probml_utils` and we also handle the case of the missing module inside `d2l.py` only.

@AnandShegde you can update the code as follows
1. remove `!wget...d2l.py`
2. Replace your imports with following,
```py
try:
    from probml_utils import d2l
except ModuleNotFoundError:
    %pip install git+https://github.com/probml/probml-utils.git
    from probml_utils import d2l
```

AnandShegde commented:
I have updated the file ask @karm-patel had suggested.

1. removed the `!wget...d2l.py`
2. Changed the import statements as suggested.
3. removed the `!mkdir figures`
